plugin/src/test/kotlin/com/exactpro/th2/gradle/ComponentTh2PluginTest.kt

```diff
@@ -66,4 +75,50 @@ class ComponentTh2PluginTest {
             },
         )
     }
+
+    @Test
+    fun `main class is specified`() {
+        val project =
+            ProjectBuilder
+                .builder()
+                .build()
+        project.version = "1.0.0"
+        project.pluginManager.apply(ComponentTh2Plugin::class.java)
+        project.the<JavaApplication>().mainClass.set("com.exactpro.th2.Main")
+
+        // We need to think how to reorganize plugins to avoid calling evaluation in tests
+        (project as DefaultProject).evaluate()
+    }
+
+    @Test
+    fun `main class isn't specified`() {
+        val project =
+            ProjectBuilder
+                .builder()
+                .build()
+        project.version = "1.0.0"
+        project.pluginManager.apply(ComponentTh2Plugin::class.java)
+
+        assertThrows<GradleException> {
+            // We need to think how to reorganize plugins to avoid calling evaluation in tests
+            (project as DefaultProject).evaluate()
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = ["", " "])
+    fun `main class has incorrect value`(mainClass: String) {
+        val project =
+            ProjectBuilder
+                .builder()
+                .build()
+        project.version = "1.0.0"
+        project.pluginManager.apply(ComponentTh2Plugin::class.java)
+        project.the<JavaApplication>().mainClass.set(mainClass)
+
+        assertThrows<GradleException> {
+            // We need to think how to reorganize plugins to avoid calling evaluation in tests
+            (project as DefaultProject).evaluate()
+        }
+    }
```

Comment:
suggestion: if we move those tests into `functionalTest` we won't need to execute evaluate on project